src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/ProtectionType.java

```diff
@@ -0,0 +1,17 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+/**
+ * Represents the {@link ProtectionType} that a {@link CustomProtection}
+ * prevents the damage from.
+ *
+ * @author Linox
+ *
+ * @see CustomProtection
+ *
+ */
+public enum ProtectionType {
```

Comment:
Not sure I'm a fan of this being an enum. I'd rather addons be able to add their own as they wish. This could be a Predicate with the damage event (or such object) and then it's tested.

Reply:
I don't think I understand this.

Reply:
Instead of accepting an enum, having like a canProtect(Predicate<Event>) and then test the predicate for if it can protect.

This way, other addons can actually add their own protection with this system.

Reply:
I agree that it would be really awesome if addons could add their own types.
But I don't really see how a Predicate would be useful here.
I would rather suggest a Namespaced-key based approach here too.

However I would also recommend to delay this discussion, we can always change it and improve this API later.
We probably bombarded Linox with change requests more than enough for now :LUL:
I'd say this is ready to be merged and we can always make the API more accessible later down the line.

Reply:
I'm ok with this.